emperor/support_files/js/emperor.js

```diff
@@ -125,136 +125,137 @@ function toggleScaleCoordinates(element) {
+	g_zMaximumValue = operation(g_zMaximumValue, g_fractionExplained[g_viewingAxes[2]]);
+	g_xMinimumValue = operation(g_xMinimumValue, g_fractionExplained[g_viewingAxes[0]]);
+	g_yMinimumValue = operation(g_yMinimumValue, g_fractionExplained[g_viewingAxes[1]]);
+	g_zMinimumValue = operation(g_zMinimumValue, g_fractionExplained[g_viewingAxes[2]]);
 	g_maximum = operation(g_maximum, g_fractionExplained[g_viewingAxes[0]])
 
 	// scale the position of the light
 	g_sceneLight.position.set(
 		operation(g_sceneLight.position.x, g_fractionExplained[g_viewingAxes[0]]),
 		operation(g_sceneLight.position.y, g_fractionExplained[g_viewingAxes[0]]),
-		operation(g_sceneLight.position.z, g_fractionExplained[g_viewingAxes[0]]));
+		operation(g_sceneLight.position.z, g_fractionExplained[g_viewingAxes[0]])
+  );
 
 	// scale the position of the camera according to pc1
 	g_sceneCamera.position.set(
 		operation(g_sceneCamera.position.x, g_fractionExplained[g_viewingAxes[0]]),
 		operation(g_sceneCamera.position.y, g_fractionExplained[g_viewingAxes[0]]),
-		operation(g_sceneCamera.position.z, g_fractionExplained[g_viewingAxes[0]]))
+		operation(g_sceneCamera.position.z, g_fractionExplained[g_viewingAxes[0]])
+  );
 	// scale the axis lines
 	drawAxisLines();
 
 	// set the new position of each of the sphere objects
-	for (sample_id in g_plotSpheres){
-		// scale the position of the spheres
-		g_plotSpheres[sample_id].position.set(
-			operation(g_plotSpheres[sample_id].position.x,g_fractionExplained[g_viewingAxes[0]]),
-			operation(g_plotSpheres[sample_id].position.y,g_fractionExplained[g_viewingAxes[1]]),
-			operation(g_plotSpheres[sample_id].position.z,g_fractionExplained[g_viewingAxes[2]]));
-	}
+  _.each(g_plotSpheres, function(sphere){
+    // scale the position of the spheres
+    sphere.position.set(
+			operation(sphere.position.x, g_fractionExplained[g_viewingAxes[0]]),
+			operation(sphere.position.y, g_fractionExplained[g_viewingAxes[1]]),
+			operation(sphere.position.z, g_fractionExplained[g_viewingAxes[2]])
+    );
+  });
 
 	// ellipses won't always be available hence the two separate loops
-	for (sample_id in g_plotEllipses){
-		// scale the dimensions of the positions of each ellipse
-		g_plotEllipses[sample_id].position.set(
-			operation(g_plotEllipses[sample_id].position.x, g_fractionExplained[g_viewingAxes[0]]),
-			operation(g_plotEllipses[sample_id].position.y, g_fractionExplained[g_viewingAxes[1]]),
-			operation(g_plotEllipses[sample_id].position.z, g_fractionExplained[g_viewingAxes[2]]));
-
-		// scale the dimensions of the ellipse
-		g_plotEllipses[sample_id].scale.set(
-			operation(g_plotEllipses[sample_id].scale.x, g_fractionExplained[g_viewingAxes[0]]),
-			operation(g_plotEllipses[sample_id].scale.y, g_fractionExplained[g_viewingAxes[1]]),
-			operation(g_plotEllipses[sample_id].scale.z, g_fractionExplained[g_viewingAxes[2]]));
-	}
+  _.each(g_plotEllipses, function(ellipse){
```

Comment:
:+1: if it is readable and gives a speed boost, then I'd think we'd definitely want this!

I think there are a few places that this will tricky to implement.  For example [drawTaxa](https://github.com/biocore/emperor/blob/master/emperor/support_files/js/emperor.js#L1448).

From what I understand, we're treating associative arrays as dictionaries - meaning that the lookup time is linear.
So it would be nice to minimize [these sorts](https://github.com/biocore/emperor/blob/master/emperor/support_files/js/emperor.js#L1458) of indexing operations